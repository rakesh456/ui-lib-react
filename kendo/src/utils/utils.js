
// import moment from 'moment';
export function getDateMMDDYYYY(date) {
    let d = new Date(date),
        month = '' + (d.getMonth() + 1),
        day = '' + d.getDate(),
        year = d.getFullYear();

    if (month.length < 2) month = '0' + month;
    if (day.length < 2) day = '0' + day;

    return [month, day, year].join('/');
}

export function getDateByFormatDDMMYYYY(date, format){
    return (format && format === 'MM/DD/YYYY')? getDateMMDDYYYY(date) : getDateMMDDYYYY(date);
}

export function  addDays(date, days) {
    var result = new Date(date);
    result.setDate(result.getDate() + days);
    return result;
}