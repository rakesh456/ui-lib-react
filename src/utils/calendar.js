export const CURRENT_YEAR = +(new Date().getFullYear());

export const CURRENT_MONTH = +(new Date().getMonth()) + 1;

export const WEEK_COUNT = 6;

export const zeroPad = (value, length) => {
    return `${value}`.padStart(length, '0');
}

export const getMonthDays = (month = CURRENT_MONTH, year = CURRENT_YEAR) => {
    const months30 = [4, 6, 9, 11];
    const leapYear = year % 4 === 0;

    return month === 2
        ? leapYear
            ? 29
            : 28
        : months30.includes(month)
            ? 30
            : 31;
}


export const getPreviousMonth = (month, year) => {
    const prevMonth = (month > 1) ? month - 1 : 12;
    const prevMonthYear = (month > 1) ? year : year - 1;

    return { month: prevMonth, year: prevMonthYear };
}

export const getNextMonth = (month, year) => {
    const nextMonth = (month < 12) ? month + 1 : 1;
    const nextMonthYear = (month < 12) ? year : year + 1;

    return { month: nextMonth, year: nextMonthYear };
}

export const getMonthFirstDay = (month = CURRENT_MONTH, year = CURRENT_YEAR) => {
    return +(new Date(`${year}-${zeroPad(month, 2)}-01`).getDay()) + 1;
}

export default (month = CURRENT_MONTH, year = CURRENT_YEAR) => {
    const monthDays = getMonthDays(month, year);
    const monthFirstDay = getMonthFirstDay(month, year);

    const daysFromPrevMonth = monthFirstDay - 1;
    const daysFromNextMonth = (WEEK_COUNT * 7) - (daysFromPrevMonth + monthDays);

    const { month: prevMonth, year: prevMonthYear } = getPreviousMonth(month, year);
    const { month: nextMonth, year: nextMonthYear } = getNextMonth(month, year);

    const prevMonthDays = getMonthDays(prevMonth, prevMonthYear);

    const prevMonthDates = [...new Array(daysFromPrevMonth)].map((n, index) => {
        const day = index + 1 + (prevMonthDays - daysFromPrevMonth);
        return [prevMonthYear, zeroPad(prevMonth, 2), zeroPad(day, 2)];
    });

    const thisMonthDates = [...new Array(monthDays)].map((n, index) => {
        const day = index + 1;
        return [year, zeroPad(month, 2), zeroPad(day, 2)];
    });

    const nextMonthDates = [...new Array(daysFromNextMonth)].map((n, index) => {
        const day = index + 1;
        return [nextMonthYear, zeroPad(nextMonth, 2), zeroPad(day, 2)];
    });

    return [...prevMonthDates, ...thisMonthDates, ...nextMonthDates];

}

// (bool) Checks if a value is a date - this is just a simple check
export const isDate = date => {
    const isDate = Object.prototype.toString.call(date) === '[object Date]';
    const isValidDate = date && !Number.isNaN(date.valueOf());

    return isDate && isValidDate;
}

// (bool) Checks if two date values are of the same month and year
export const isSameMonth = (date, basedate = new Date()) => {

    if (!(isDate(date) && isDate(basedate))) return false;

    const basedateMonth = +(basedate.getMonth()) + 1;
    const basedateYear = basedate.getFullYear();

    const dateMonth = +(date.getMonth()) + 1;
    const dateYear = date.getFullYear();

    return (+basedateMonth === +dateMonth) && (+basedateYear === +dateYear);
}

// (bool) Checks if two date values are the same day
export const isSameDay = (date, basedate = new Date()) => {
    if (!(isDate(date) && isDate(basedate))) return false;
    const basedateDate = basedate.getDate();
    const basedateMonth = +(basedate.getMonth()) + 1;
    const basedateYear = basedate.getFullYear();

    const dateDate = date.getDate();
    const dateMonth = +(date.getMonth()) + 1;
    const dateYear = date.getFullYear();

    return (basedateDate === dateDate) && (basedateMonth === dateMonth) && (basedateYear === dateYear);
}

// (string) Formats the given date as YYYY-MM-DD
// Months and Days are zero padded
export const getIsoDate = (date = new Date()) => {
    if (!isDate(date)) return null;
    return [
        date.getFullYear(),
        zeroPad(+date.getMonth() + 1, 2),
        zeroPad(+date.getDate(), 2)
    ].join('-');
}

