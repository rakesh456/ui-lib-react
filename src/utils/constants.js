export const KEY_CODES = {
    ESCAPE:  27,
    SPACE:  32,
    COMMA:  44,
    NINE:  57,
}