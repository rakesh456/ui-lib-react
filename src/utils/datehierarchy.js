
export const getMonthDays = (month, year) => {
  const months30 = [4, 6, 9, 11];
  const leapYear = year % 4 === 0;
  let day =  month === 2
      ? leapYear
          ? 29
          : 28
      : (months30.indexOf(month) !== -1)
          ? 30
          : 31;
          var days = [];
          for(let i = 1 ; i <= day ; i++){
            var dayObj = {day :i,state : 0}
            days.push(dayObj);
          }
          return days;
}
export const getChildren = function (year, showWeeks) {
  if(showWeeks === false){
  
  var quarterArray = [{
  "quarter": "Q1-" + year,
  "showChild": false,
  "state": 0,
  "children": [{
    "month": "Jan-"+year,
    "showChild": false,
    "state": 0,
    "children": getMonthDays(1, year),
   },
   {
    "month": "Feb-"+year,
    "showChild": false,
    "state": 0,
    "children":getMonthDays(2, year),

   },
   {
    "month": "Mar-"+year,
    "showChild": false,
    "state": 0,
    "children": getMonthDays(3, year)
   }
  ]
 },
 {
  "quarter": "Q2-"+year,
  "showChild": false,
  "state": 0,
  "children": [{
    "month": "Apr-"+year,
    "showChild": false,
    "state": 0,
    "children":getMonthDays(4, year)
   },
   {
    "month": "May-"+year,
    "showChild": false,
    "state": false,
    "children":getMonthDays(5, year)
   },
   {
    "month": "Jun-"+year,
    "showChild": false,
    "state": 0,
    "children": getMonthDays(6, year)
   }
  ]
 },
 {
  "quarter": "Q3-"+year,
  "showChild": false,
  "state": 0,
  "children": [{
    "month": "Jul-"+year,
    "showChild": false,
    "state": 0,
    "children": getMonthDays(7, year)
   },
   {
    "month": "Aug-"+year,
    "showChild": false,
    "state": 0,
    "children": getMonthDays(8, year)
   },
   {
    "month": "Sep-"+year,
    "showChild": false,
    "state": 0,
    "children": getMonthDays(9, year)
   }
  ]
 },
 {
  "quarter": "Q4-"+year,
  "showChild": false,
  "state": 0,
  "children": [{
    "month": "Oct-"+year,
    "showChild": false,
    "state": 0,
    "children": getMonthDays(10, year)
   },
   {
    "month": "Nov-"+year,
    "showChild": false,
    "state": 0,
    "children": getMonthDays(11, year)
   },
   {
    "month": "Dec-"+year,
    "showChild": false,
    "state": 0,
    "children": getMonthDays(12, year)
   }
  ]
 }
]
}
else{
   quarterArray = [{
    "quarter": "Q1",
    "showChild": false,
    "state": 0,
    "children": [{
      "month": "Jan",
      "showChild": false,
      "state": 0,
      "children": getMonthWeeks(1, year)
     },
     {
      "month": "Feb",
      "showChild": false,
      "state": 0,
      "children":getMonthWeeks(2, year),
  
     },
     {
      "month": "Mar",
      "showChild": false,
      "state": 0,
      "children": getMonthWeeks(3, year)
     }
    ]
   },
   {
    "quarter": "Q2",
    "showChild": false,
    "state": 0,
    "children": [{
      "month": "Apr",
      "showChild": false,
      "state": 0,
      "children":getMonthWeeks(4, year)
     },
     {
      "month": "May",
      "showChild": false,
      "state": false,
      "children":getMonthWeeks(5, year)
     },
     {
      "month": "Jun",
      "showChild": false,
      "state": 0,
      "children": getMonthWeeks(6, year)
     }
    ]
   },
   {
    "quarter": "Q3",
    "showChild": false,
    "state": 0,
    "children": [{
      "month": "Jul",
      "showChild": false,
      "state": 0,
      "children": getMonthWeeks(7, year)
     },
     {
      "month": "Aug",
      "showChild": false,
      "state": 0,
      "children": getMonthWeeks(8, year)
     },
     {
      "month": "Sep",
      "showChild": false,
      "state": 0,
      "children": getMonthWeeks(9, year)
     }
    ]
   },
   {
    "quarter": "Q4",
    "showChild": false,
    "state": 0,
    "children": [{
      "month": "Oct",
      "showChild": false,
      "state": 0,
      "children": getMonthWeeks(10, year)
     },
     {
      "month": "Nov",
      "showChild": false,
      "state": 0,
      "children": getMonthWeeks(11, year)
     },
     {
      "month": "Dec",
      "showChild": false,
      "state": 0,
      "children": getMonthWeeks(12, year)
     }
    ]
   }
  ]
}
return quarterArray;
}
export const getListOfYears = function(lowerLimit, upperLimit, showWeeks) {
if(lowerLimit > 999 && upperLimit>999 && (lowerLimit <= upperLimit) && lowerLimit % 1 ===0 && upperLimit % 1 === 0)
{
    let years = [];
    while (lowerLimit <= upperLimit) {
     var year = {
      "year":parseInt(lowerLimit),
      "showChild": false,
      "state": 0,
      "children": getChildren(lowerLimit, showWeeks)
     }
     years.push(year);
     lowerLimit++;
    }
    return years;
   }
  
  else
  {
    lowerLimit = 2;
    upperLimit = 1;
    let years = [];
    while (lowerLimit <= upperLimit) {
      year = {
      "year": lowerLimit,
      "showChild": false,
      "state": 0,
      "children": getChildren(lowerLimit)
     }
     years.push(year);
     lowerLimit++;
    }
    console.error("either lowerLimit or upperLimit is not in valid format");
    return years;
   
    
  }
}



  export const getMonthWeeks = function (month_number,year ) {

    var firstOfMonth = new Date(year, month_number-1, 1);
    var lastOfMonth = new Date(year, month_number, 0);
    var used = firstOfMonth.getDay() + lastOfMonth.getDate();
    var weeks = [];
    var start = 1;
    var weekdays = new Array(7);
    weekdays[0] = "Sun";
    weekdays[1] = "Mon";
    weekdays[2] = "Tue";
    weekdays[3] = "Wed";
    weekdays[4] = "Thu";
    weekdays[5] = "Fri";
    weekdays[6] = "Sat";
    for(let weekNo = 1; weekNo <=  Math.ceil(used/7); weekNo++){
      var weekobj = { week : "Week "+ weekNo, state: 0, showChild: false, children:[]}
      for(var i = start; i < (lastOfMonth).getDate()+1;i++ ){
          var monthDate = new Date(year, month_number-1, i);
          var dayObj = {date: i, day: weekdays[monthDate.getDay()], state: 0}; 
          weekobj.children.push(dayObj);
          if(monthDate.getDay()===6){
              start = i + 1;
              break;
          }
      }
      weeks.push(weekobj);
    }
    return weeks;
}
