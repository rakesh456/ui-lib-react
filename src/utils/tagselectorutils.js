export const DEFAULT_OPTIONS = {'placeholder': 'Search', 'listItems': [], 'allowNewValue': false, 'maxItemCounter': 0, 'searchWithHelper': false, 'readOnly': false, 'allowHierarchy': false};

// Function to reset options with default options
export const resetTagSelectorOptions = (options) => {
    return {...DEFAULT_OPTIONS, ...options};
}

// Function to reset options with default options
export const isValidJsonFormat = (allowHierarchy, json) => {
    let isValidFormat = true;
    let key;
    if(!json || json.length <= 0){
        isValidFormat = false;
    } else {
        if(allowHierarchy === true){
            json.forEach((item) => {
                for (key in item) {
                    isValidFormat = (typeof key === 'string' && typeof item[key] === 'object')
                }
            });
        } else {
            json.forEach((item) => {
                for (key in item) {
                    isValidFormat = (typeof key === 'string' && typeof item[key] === 'string')
                }
            });
        }
    }
    return isValidFormat;
}

// Function to reset options with default options
export const sortListingByType = (allowHierarchy, json) => {
    if(!json || json.length <= 0){
        return [];
    } else {
        if(allowHierarchy === true){
            // json.sort(function (a, b) {
            //     return a.value - b.value;
            // });
        } else {
            // let data = json.sort(function (a, b) {
            //     return b.value - a.value;
            // });
            // console.log(' data ', JSON.stringify(data));
        }
    }
    return json;
}
