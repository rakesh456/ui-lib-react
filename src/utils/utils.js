import { isDate } from "./calendar";

// import moment from 'moment';
export const Fragment = (props, children) => children;

// Calendar months names
export const MONTH_NAMES = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
export const WEEK_SHORT_NAMES = ["S", "M", "T", "W", "T", "F", "S"];

export function isUndefinedOrNull(obj){
    return (typeof obj === "undefined" || obj === null || !obj)? true : false;
}

export function getMonthNameByIndex(index) {
    var _index = (index)? index : 0;
    return MONTH_NAMES[_index].toUpperCase();
}

export function splitArray(array, chunk_size){
    var index = 0;
    var arrayLength = array.length;
    var tempArray = [];
    
    for (index = 0; index < arrayLength; index += chunk_size) {
        const myChunk = array.slice(index, index+chunk_size);
        tempArray.push(myChunk);
    }

    return tempArray;
}

export function getStaticDays(){
    return [[2019,"07","28"],[2019,"07","29"],[2019,"07","30"],[2019,"07","31"],[2019,"08","01"],[2019,"08","02"],[2019,"08","03"],[2019,"08","04"],[2019,"08","05"],[2019,"08","06"],[2019,"08","07"],[2019,"08","08"],[2019,"08","09"],[2019,"08","10"],[2019,"08","11"],[2019,"08","12"],[2019,"08","13"],[2019,"08","14"],[2019,"08","15"],[2019,"08","16"],[2019,"08","17"],[2019,"08","18"],[2019,"08","19"],[2019,"08","20"],[2019,"08","21"],[2019,"08","22"],[2019,"08","23"],[2019,"08","24"],[2019,"08","25"],[2019,"08","26"],[2019,"08","27"],[2019,"08","28"],[2019,"08","29"],[2019,"08","30"],[2019,"08","31"],[2019,"09","01"],[2019,"09","02"],[2019,"09","03"],[2019,"09","04"],[2019,"09","05"],[2019,"09","06"],[2019,"09","07"]];
}

export function getFormatfromOptions(options){
    return (!isUndefinedOrNull(options) && options.displayFormat)? options.displayFormat : 'MM/DD/YYYY';
}

export function getDateByFormatDDMMYYYY(date, format){
    return (format && format === 'DD/MM/YYYY')? getDateDDMMYYYY(date) : getDateMMDDYYYY(date);
}

export const isValidDate = dateObject => { 
    return new Date(dateObject).toString() !== 'Invalid Date'; 
}

export function getDateDDMMYYYY(date, format) {
    // return getDateMMDDYYYY(date);
    // let d = new Date(date);
    // let d = (isDate(date))? new Date(date) : new Date(convertYYYYMMDD(date, {}));
    let d = (isValidDate(date))? new Date(date) : new Date(convertYYYYMMDDByFormat(date, 'DD/MM/YYYY'));
    
    let month = '' + (d.getMonth() + 1),
        day = '' + d.getDate(),
        year = d.getFullYear();

    if (month.length < 2) month = '0' + month;
    if (day.length < 2) day = '0' + day;

    return [day, month, year].join('/');
}

export function getDateMMDDYYYY(date) {
    let d = new Date(date),
        month = '' + (d.getMonth() + 1),
        day = '' + d.getDate(),
        year = d.getFullYear();

    if (month.length < 2) month = '0' + month;
    if (day.length < 2) day = '0' + day;

    return [month, day, year].join('/');
}

export function convertYYYYMMDD(date, options) {
    let format = getFormatfromOptions(options);
    return convertYYYYMMDDByFormat(date, format);
}

export function convertYYYYMMDDByFormat(date, format){
    let dayIndex = (format && format === 'MM/DD/YYYY')? 1 : 0;
    let monthIndex = (format && format === 'MM/DD/YYYY')? 0 : 1;
    let d = date.toString().split("/"),
        month = '' + d[monthIndex],
        day = '' + d[dayIndex],
        year = d[2];

    if (month.length < 2) month = '0' + month;
    if (day.length < 2) day = '0' + day;

    return [year, month, day].join('-');
}

export function getCurrentYear() {
    return new Date().getFullYear();
}

export function isValidFormattedDate(date, options) {
    let _date = convertYYYYMMDD(date, options);
    return (isDate(new Date(_date)));
}

export function guid() {
    return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
        s4() + '-' + s4() + s4() + s4();
}

export function s4() {
    return Math.floor((1 + Math.random()) * 0x10000)
        .toString(16)
        .substring(1);
}