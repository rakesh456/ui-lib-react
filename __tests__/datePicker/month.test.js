
import Month from '../../src/components/Datepicker/calendar-date/month';
import DatePicker from '../../src/components/Datepicker/index';
import DEFAULT_OPTIONS from '../../src/utils/constants';
import { isUndefinedOrNull, isBlank, splitArray, getFormatfromOptions, getDateByFormat, isValidDate, getDateDDMMYYYYNew, getDateMMDDYYYY, getYYYYMMDD, dateToYear, dateToMMYYYY, dateToQQYYYY, getQQFromMonth, isObject, isStringExists, compareObjects } from '../../src/utils/utils.js';
import { zeroPad, MONTH_NAMES } from '../../src/utils/calendar';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import React from 'react';
import Days from '../../src/components/Datepicker/calendar-date/days';
import { FaCaretRight } from 'react-icons/lib/fa';


// 0. checking month component is rendering properly.
 
test('Month component renders the month correctly', () => {
  const options = DEFAULT_OPTIONS;
  const rendered = renderer.create(
    <Month options={options} />
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});


// 1. Test for isUndefinedOrNull

describe('Testing on method isUndefinedOrNull', () => {
  let dateTestValuesInputArray = [123, undefined, null]
  let dateTestValuesOutputArray = [false, true, true]
  let messageDisplayForEachTestCase = ['If any defined variable or truthy value is passed then function will return False',
    'If Undefined value is passed then function returns true',
    'If null value is passed then function returns true']
  for (let i = 0; i < dateTestValuesInputArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {

      expect(isUndefinedOrNull(dateTestValuesInputArray[i]))
        .toEqual(dateTestValuesOutputArray[i])
    })

  }

})


// 2. Test for isObject

describe('Testing on function isObject', () => {
  let obj = {}
  let obj1;
  let obj2 = null;
  let dateTestValuesInputArray = [obj, obj1, obj2]
  let dateTestValuesOutputArray = [true, false, false]
  let messageDisplayForEachTestCase = ['Passing object function returns true ', 'Passing undefined object then function returns false', 'Passing null object then function returns false']
  for (let i = 0; i < dateTestValuesInputArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {
      expect(isObject(dateTestValuesInputArray[i])).toEqual(dateTestValuesOutputArray[i])
    })
  }
})


// 3. Test for isBlank

describe('Testing on function isBlank', () => {
  let dateTestValuesInputArray = ['JSTIGERS', undefined, null, '']
  let dateTestValuesOutputArray = [false, true, true, true]
  let messageDisplayForEachTestCase = ['If any defined variable or truthy value is passed then function will return False', 'If Undefined value is passed then function returns true', 'If null value is passed then function returns true']
  for (let i = 0; i < dateTestValuesInputArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {
      expect(isBlank(dateTestValuesInputArray[i])).toEqual(dateTestValuesOutputArray[i])
    })

  }
})


// 4. Test for splitArray

describe('Testing on function splitArray', () => {
  let splitTestValuesInputArray = [[1, 2, 3, 4, 5, 6, 7, 8, 9, 10], [1, 2, 3, 4, 5, 6, 7, 8, 9, 10], []]
  let splitTestValuesOutputArray = [[[1, 2], [3, 4], [5, 6], [7, 8], [9, 10]], [[1, 2, 3, 4, 5], [6, 7, 8, 9, 10]], []]
  let chunkingSizeofInputArray = [2, 5, 2]
  let messageDisplayForEachTestCase = ['If any valid date is passed to the function it return true', 'If any Invalid date is passed to the function it return false']
  for (let i = 0; i < splitTestValuesInputArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {

      expect(splitArray(splitTestValuesInputArray[i], chunkingSizeofInputArray[i])).toEqual(splitTestValuesOutputArray[i])
    })

  }
})


// 5. Test for getFormatfromOptions that converts any date formate into 'MM/DD/YYYY'

describe('Testing on function getFormatfromOption', () => {
  let dateTestValuesInputArray = ['YYYY/MM/DD', 'DD/MM/YYYY', undefined, null]
  let dateTestValuesOutputArray = ['MM/DD/YYYY', 'MM/DD/YYYY', 'MM/DD/YYYY', 'MM/DD/YYYY']
  let messageDisplayForEachTestCase = ['If any format of date is passed to a function it returns in as specified in option format', 'If change the format of date is passed to a function it returns in as specified in option format', 'If Undefined value is passed then function returns in MM/DD/YYYY format ', 'If null value is passed then function returns in MM/DD/YYYY format']
  for (let i = 0; i < dateTestValuesInputArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {
      expect(getFormatfromOptions(dateTestValuesInputArray[i])).toEqual(dateTestValuesOutputArray[i])
    })
  }
})


// 6. Test for getDateByFormat

describe('Testing on function getDateByFormat ', () => {
  let dateTestValuesInputArray = ['12/01/2019', '12/01/2019', '12/01/2019', '12/01/2019']
  let dateTestValueInputFormatArray = ['DD/MM/YYYY', 'MM/DD/YYYY', undefined, null]
  let dateTestValuesOutputArray = ['01/12/2019', '12/01/2019', '12/01/2019', '12/01/2019']
  let messageDisplayForEachTestCase = ['If any date is passed in DD/MM/YYYY format to the function it returns in the format same as given', 'If any date is passed in MM/DD/YYYY format to the function it returns in the default format MM/DD/YYYY ', 'If for any date, undefined value is passed as format then function returns in default format MM/DD/YYYY', 'If for any date, null value is passed as format then function returns in default format MM/DD/YYYY']
  for (let i = 0; i < dateTestValuesInputArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {
      expect(getDateByFormat(dateTestValuesInputArray[i], dateTestValueInputFormatArray[i])).toEqual(dateTestValuesOutputArray[i])
    })
  }
})


// 7. isValid

describe('Testing on function isValid', () => {
  let dateTestValuesInputArray = ['12/01/2019', '18/01/2019']
  let dateTestValuesOutputArray = [true, false]
  let messageDisplayForEachTestCase = ['If any valid date is passed to the function it return true', 'If any Invalid date is passed to the function it return false']
  for (let i = 0; i < dateTestValuesInputArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {
      expect(isValidDate(dateTestValuesInputArray[i])).toEqual(dateTestValuesOutputArray[i])
    })
  }
})


// 8. Tast for getDateDDMMYYYYNew

describe('Testing on function getDateDDMMYYYYNew', () => {
  let dateTestValuesInputArray = ['3/4/2019', '2019/03/05', undefined, '', null]
  let dateTestValuesOutputArray = ['03/04/2019', '03/05/2019', 'NaN/NaN/NaN', 'NaN/NaN/NaN', 'NaN/NaN/NaN']
  let messageDisplayForEachTestCase = ['function getDateDDMMYYYYNew give format in DD/MM/YYYY', '']
  for (let i = 0; i < dateTestValuesInputArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {

      expect(getDateDDMMYYYYNew(dateTestValuesInputArray[i])).toEqual(dateTestValuesOutputArray[i])
    })

  }

})


// 9. Test for getDateMMDDYYYY

describe('Testing on function getDateMMDDYYYY', () => {
  let dateTestValuesInputArray = ['3/4/2019', '18/01/2019', '2019/02/03', '02/2019/03',]
  let dateTestValuesOutputArray = ['03/04/2019', 'NaN/NaN/NaN', '02/03/2019', 'NaN/NaN/NaN']
  let messageDisplayForEachTestCase = ['If we pass date the function return in MM/DD/YYYY format', 'If we pass invalid date returns Not a Number', 'If we pass date in different format YYYY/MM/DD it converts to MM/DD/YYYY', 'If we pass date DD/YYYY/MM it returns Not a number']
  for (let i = 0; i < dateTestValuesInputArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {
      expect(getDateMMDDYYYY(dateTestValuesInputArray[i])).toEqual(dateTestValuesOutputArray[i])
    })
  }
})


// 10. Test for getYYYYMMDD

describe('Testing on funtion getYYYYMMDD', () => {
  test('If we pass date then the function getYYYYMMDD return date in YYYYMMDD format', () => {
    let date = '02/05/2019'
    expect(getYYYYMMDD(date)).toEqual('2019-02-05')
  })
})


// 11. Test for dateToYear

describe('Testing on function dateToYear', () => {
  let d = new Date()
  let dateTestValuesInputArray = ['3/4/1987', '18/01/2019', '2015/02/03', null, undefined]
  let dateTestValuesOutputArray = [1987, d.getFullYear(), 2015, d.getFullYear(), d.getFullYear()]
  let messageDisplayForEachTestCase = ["If we pass date to function it return it's full year", 'If we pass invalid date function returns current date full year ', "If we pass date in different format then also it gives it's full year", "For null value function will return current date fully year", "For udefined value function will return current date fully year"]
  for (let i = 0; i < dateTestValuesInputArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {
      expect(dateToYear(dateTestValuesInputArray[i])).toEqual(dateTestValuesOutputArray[i])
    })
  }
})


// 12. Test of function dateToMMYYYY

describe('Testing on function dateToMMYYYY', () => {
  let d = new Date()
  let outputDate = dateFormat(d)
  function dateFormat(date) {
    let month = '' + (1 + date.getMonth())
    if (month.length < 2) month = '0' + month;
    let year = '' + date.getFullYear();
    date = [month, year].join('/')
    return date
  }
  let dateTestValuesInputArray = ['3/4/1987', '18/01/2019', '2015/02/03', null, undefined]
  let dateTestValuesOutputArray = ['03/1987', outputDate, '02/2015', outputDate, outputDate]
  let messageDisplayForEachTestCase = ["If we pass date to function it return it's months and full year of the date in MM/YYYY format", 'If we pass invalid date function returns current date month and full year in MM/YYYY format', "If we pass date in different format then also it gives it's month and full year date in MM/YYYY format", "For null value function will return current date month and fully year in MM/YYYY format", "For udefined value function will return current date month and fully year in MM/YYYY format"]
  for (let i = 0; i < dateTestValuesInputArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {
      expect(dateToMMYYYY(dateTestValuesInputArray[i])).toEqual(dateTestValuesOutputArray[i])
    })
  }
})


// 13. Test of function dateTo QQYYYY

describe('Testing on function dateToQQYYYY', () => {
  let d = new Date()
  let currentMonth = d.getMonth() + 1
  let _val = (currentMonth <= 12 && currentMonth >= 10) ? 4 : (currentMonth <= 9 && currentMonth >= 7) ? 3 : (currentMonth <= 6 && currentMonth >= 4) ? 2 : 1;
  let quarter = 'Q' + _val;
  let dateTestValuesInputArray = ['3/4/1987', '18/01/2019', '2015/02/03', null, undefined]
  let dateTestValuesOutputArray = ['Q1/1987', quarter + '/' + d.getFullYear(), 'Q1/2015', quarter + '/' + d.getFullYear(), quarter + '/' + d.getFullYear()]
  let messageDisplayForEachTestCase = ["If we pass date to function it return Quarter with its full year", 'If we pass invalid date function returns current month Quarter and its full year ', "If we pass date in different format then also it gives it's current quarter and full year", "For null value function will return current month quarter and fully year", "For udefined value function will return current month quarter and  fully year"]
  for (let i = 0; i < dateTestValuesInputArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {
      expect(dateToQQYYYY(dateTestValuesInputArray[i])).toEqual(dateTestValuesOutputArray[i])
    })
  }
})


// 14. Test of function getQQFromMonth

describe('Testing of function getQQFromMonth', () => {
  let d = 5
  test('Testing a function that it return the quarter of the date ', () => {
    expect(getQQFromMonth(d)).toEqual('Q2')
  })
  let month = 15
  test('Testing a function that it return the quarter of the date ', () => {
    expect(getQQFromMonth(month)).toEqual('Q1')
  })
})


// 16. Test of isStringExist

describe('Testing on function isStringExists', () => {
  let TestValuesInputStringArray = ['amanverma', 'thisisbengaluru']
  let TestValuesInputSubStringArray = ['aman', 'balloon']
  let TestValuesOutputArray = [true, false]
  let messageDisplayForEachTestCase = ['Passing string and substring if substring present then function return true', 'Passing string and substring if substring not present then function return false', 'If null value is passed then function returns true']
  for (let i = 0; i < TestValuesInputStringArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {
      expect(isStringExists(TestValuesInputStringArray[i], TestValuesInputSubStringArray[i])).toEqual(TestValuesOutputArray[i])
    })
  }
})


// 17. Test for compareObjects

describe('Testing on function compareObjects', () => {
  let person = { firstName: "John", lastName: "Doe", age: 50, eyeColor: "blue" };
  let otherPerson = { firstName: "John", lastName: "Doe", age: 50, eyeColor: "blue" };
  let differentValuePerson = { firstName: "suraj", lastName: "verma", age: 30, eyeColor: "brown" };
  let differentPropertyPerson = { hairColor: 'brown', tall: '6.0ft' }
  let emptyObject = {}
  let emptyObject2 = {}
  let TestValuesInputObjectArray = [person, person, person, emptyObject, person]
  let TestValuesInputOtherObjectArray = [otherPerson, differentValuePerson, emptyObject, emptyObject2, differentPropertyPerson]
  let TestValuesOutputArray = [true, false, false, true, false]
  let messageDisplayForEachTestCase = ['Passing two objects having same property and values then the function return true', 'Passing two different  objects having same property and different values then the function return flase', 'Comparing Object to an empty object then the function returns false', "Compare two empty objects then the funtions returns true", "Compare two objects having different property then the function return false"]
  for (let i = 0; i < TestValuesInputObjectArray.length; i++) {
    test(`${messageDisplayForEachTestCase[i]}`, () => {
      expect(compareObjects(TestValuesInputObjectArray[i], TestValuesInputOtherObjectArray[i])).toEqual(TestValuesOutputArray[i])
    })
  }

})


describe('Should render without crashing', () => {
  let wrapper;
  test('Check month rendering correctly', () => {
    wrapper = shallow(<Month month='12' year='2019' />);
    let html = wrapper.find('.VS-MonthName').text();
    expect(html).toEqual('DECEMBER 2019');
  })
})