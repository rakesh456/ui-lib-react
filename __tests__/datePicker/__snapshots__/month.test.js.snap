// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month component renders the month correctly 1`] = `
<div
  className="VS-CalendarMonth VS-TextCenter"
>
  <svg
    className="VS-PullLeft VS-Icon"
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m26.3 10v20q0 0.6-0.4 1t-1 0.4-1-0.4l-10-10q-0.5-0.4-0.5-1t0.5-1l10-10q0.4-0.4 1-0.4t1 0.4 0.4 1z"
      />
    </g>
  </svg>
  <span
    className="VS-Medium-UPPER-Case VS-MonthName"
  >
    JANUARY
     
  </span>
  <svg
    className="VS-PullRight VS-Icon"
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m26.4 20q0 0.6-0.5 1l-10 10q-0.4 0.4-1 0.4t-1-0.4-0.4-1v-20q0-0.6 0.4-1t1-0.4 1 0.4l10 10q0.5 0.4 0.5 1z"
      />
    </g>
  </svg>
</div>
`;
